src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/WinRTSpellerInteropExtensions.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using MS.Internal.WindowsRuntime.Windows.Data.Text;
+using System.Collections.Generic;
+
+using System.Windows.Documents.MsSpellCheckLib;
+using System.Windows.Documents.Tracing;
+using static System.Windows.Documents.WinRTSpellerInterop;
+
+namespace System.Windows.Documents
+{
+    internal static class WinRTSpellerInteropExtensions
+    {
+        /// <summary>
+        /// Tokenizes <paramref name="text"/> using <paramref name="segmenter"/>, and then identifies fixes-up
+        /// the tokens to account for any missed text "in-between" those tokens. 
+        /// </summary>
+        /// <param name="segmenter">Word-breaker instance</param>
+        /// <param name="text">The text being tokenized</param>
+        /// <param name="spellChecker">The spell-checker instance used to augment the tokenizing process</param>
+        /// <param name="owner">Calling <see cref="WinRTSpellerInterop"/> instance</param>
+        /// <returns></returns>
+        /// <remarks>
+        /// Windows.Data.Text.WordsSegmenter tends to drop punctuation characters like period ('.') 
+        /// when tokenizing text. Though this behavior is compatible with a vast majority of text-processing
+        /// scenarios (like word-counting), it is not ideal for spell-checking. 
+        /// 
+        /// In this method, the following <paramref name="spellChecker"/> augmented heuristic is applied to update the token-list generated by 
+        /// <paramref name="segmenter"/>. 
+        /// 
+        ///  - Identify if any text 'missingFragment' has been dropped by the <paramref name="segmenter"/>
+        ///  - If the token immediately preceding 'missingFragment', previousToken, has a spelling error, then attempt to 
+        ///     create new candiate tokens in the following order:
+        ///     
+        ///             previousToken + missingFragment[0..0]
+        ///             previousToken + missingFragment[0..1]
+        ///             previousToken + missingFragment[0..2]
+        ///             ...
+        ///             ...
+        ///             previousToken + missingFragment[0..LEN-1], where LEN = LEN(missingFragment)
+        ///             
+        ///  - Select the first candidate token that is free of spelling errors, and replace 'previousToken' with it. 
```

Comment:
Are there any examples where there are two potential candidates and greedily selecting the first one would be incorrect?  Something like hyphenated acronyms maybe (is that a thing)?

Reply:
Should we instead be finding the last such candidate and using that?